src/main/java/me/gnat008/perworldinventory/data/DataSourceProvider.java

```diff
@@ -0,0 +1,62 @@
+package me.gnat008.perworldinventory.data;
+
+import me.gnat008.perworldinventory.DataFolder;
+import me.gnat008.perworldinventory.PerWorldInventory;
+import me.gnat008.perworldinventory.PwiLogger;
+import me.gnat008.perworldinventory.config.DatabaseProperties;
+import me.gnat008.perworldinventory.config.Settings;
+import org.bukkit.Bukkit;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.io.File;
+import java.io.IOException;
+import java.sql.SQLException;
+
+/**
+ * Creates the data source
+ */
+public class DataSourceProvider implements Provider<DataSource> {
```

Comment:
For this to take effect it needs to be registered as the provider of DataSource. Somewhere in the main class' onEnable process: `injector.registerProvider(DataSource.class, DataSourceProvider.class)` (or similar; typing this off the top of my head)